main.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"code-review-bot/internal"
+	"code-review-bot/internal/web"
+	"code-review-bot/internal/web/routes"
+	"context"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v68/github"
+	"github.com/gregjones/httpcache"
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/echo/v4/middleware"
+	"github.com/openai/openai-go"
+	"github.com/openai/openai-go/option"
+	"go.uber.org/zap"
+)
+
+func initializeGithubClient(ctx context.Context, githubToken string, logger *zap.Logger) *github.Client {
+	client := github.NewClient(httpcache.NewMemoryCacheTransport().Client()).WithAuthToken(githubToken)
+
+	user, _, err := client.Users.Get(ctx, "")
+	if err != nil {
+		logger.Fatal("Failed to authenticate with GitHub API", zap.Error(err))
+	}
+	logger.Info("Authenticated with GitHub API", zap.String("username", user.GetLogin()))
+
+	return client
+}
+
+func initializeOpenAIClient(openAIToken string, logger *zap.Logger) *openai.Client {
+	client := openai.NewClient(option.WithAPIKey(openAIToken))
+
+	logger.Info("OpenAI client initialized")
+
+	return client
+}
+
```

Comment:
Ensure that you check for nil value of `logger` after initialization. If `zap.NewDevelopment()` fails and returns a nil logger, the subsequent `logger.Fatal(...)` calls will cause a panic. You should handle the error returned by `zap.NewDevelopment()` properly, similar to how you handle errors when loading configuration.